### PR TITLE
Refactor/tree testing

### LINKED ISSUE #453: refactor: clean up tree testing

### PR DESCRIPTION
**Pull request by KristianKjerstad**
## What does this pull request change?

refactor tree testing 
* split up the monster file "test_tree_node_helpers" into several parts.
* collect all tests for tree into a new folder "test_tree_node_functionality"

Also, fix a bug where an _id is added inside a reference. THis has now been fixed.
Example:
Before, when formExample.inputEntity was updated, the document would change from
```json
{
    "_id": 'formExample',
    "type": 'dmss://DemoDataSource/plugins/form/blueprints/FormBlueprint',
    "blueprintPicker": 'FormBlueprint',
    "aNumber": 1337,
    "inputEntity": {
        "type": 'dmss://system/SIMOS/Reference',
        "referenceType": 'link',
        "address": 'dmss://DemoDataSource/$product1'
    },
}
```

to

```json
{
    "_id": 'formExample',
    "type": 'dmss://DemoDataSource/plugins/form/blueprints/FormBlueprint',
    "blueprintPicker": 'FormBlueprint',
    "aNumber": 1337,
    "inputEntity": {
        "type": 'dmss://system/SIMOS/Reference',
        "referenceType": 'link',
        "address": '$12312gads12-fsd12qwdas-12'
        "_id": "12312gads12-fsd12qwdas-12"
    },
}
```

The bug is that an extra _id field is added to the reference object, and the address is wrongly updated to be equal to this new _id value

## Issues related to this change:
closes #453

### CODE REVIEW

File: src/tests/unit/test_tree_node_helpers.py [hunk 1]
```diff
@@ -1,1087 +0,0 @@
-import unittest
```
Comment (KristianKjerstad) [inline src/tests/unit/test_tree_node_helpers.py:1]: stuff in this file has been moved to the folder "test_tree_functionality and splitted up into several files

File: src/tests/unit/test_tree_functionality/blueprints_for_tree_tests.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
+from domain_classes.blueprint import Blueprint
```
Comment (KristianKjerstad) [inline src/tests/unit/test_tree_functionality/blueprints_for_tree_tests.py:1]: this file collects blueprints that are used for several tests

File: src/tests/unit/test_tree_functionality/get_node_for_tree_tests.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+from domain_classes.tree_node import ListNode, Node
+from enums import REFERENCE_TYPES, SIMOS
+from tests.unit.mock_utils import (
+    get_mock_document_service,
+    mock_storage_recipe_provider,
+)
+from tests.unit.test_tree_functionality.blueprints_for_tree_tests import get_blueprint
+
+
+def get_engine_package_node() -> Node:
```
Comment (KristianKjerstad) [inline src/tests/unit/test_tree_functionality/get_node_for_tree_tests.py:11]: content of get_engine_package_node() is the same as in the old get_engine_package_node() inside the old file "test_tree_node_helpers.py"

File: src/tests/unit/test_tree_functionality/tree_dict_conversion.py [hunk 4]
```diff
@@ -0,0 +1,328 @@
+import unittest
```
Comment (KristianKjerstad) [inline src/tests/unit/test_tree_functionality/tree_dict_conversion.py:1]: the tests in this files are copied from the old "test_tree_node_helpers.py" file, without being modified. The only difference is that a new test "test_tree_node_to_ref_dict_2" has been added.

File: src/tests/unit/test_tree_functionality/test_tree_node_helpers.py [hunk 5]
```diff
@@ -0,0 +1,449 @@
+import unittest
```
Comment (KristianKjerstad) [inline src/tests/unit/test_tree_functionality/test_tree_node_helpers.py:1]: all testsare copied from the old "test_tree_node_helpers.py" file without being modified.

File: src/tests/unit/test_tree_functionality/test_tree_node_delete.py [hunk 6]
```diff
@@ -0,0 +1,167 @@
+import unittest
```
Comment (KristianKjerstad) [inline src/tests/unit/test_tree_functionality/test_tree_node_delete.py:1]: all tests are copied from the old "test_tree_node_helpers.py" file without being modified.

File: src/tests/unit/test_tree_functionality/test_tree_node_update.py [hunk 7]
```diff
@@ -126,6 +129,23 @@ def get_blueprint(type: str):
                     ],
                 }
             )
+        elif type == "FormBlueprint":
```
Comment (KristianKjerstad) [inline src/tests/unit/test_tree_functionality/test_tree_node_update.py:132]: new blueprint is added, that is used in the new test "test_set_update_uncontained_child"

**Review by eoaksnes — APPROVED** [review]

**Pull request merged**